### PR TITLE
Ogg dev

### PR DESCRIPTION
**Pull request by Oggal**
Implements Run_Command
Modifies Prep_Input to reuse token buffer

Frees token buffer at end of program

### CODE REVIEW

File: input.c [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
 
-	resultArray = malloc(16 * sizeof(char *));
-	if (!resultArray)
+
+	if(resultArray == NULL)
+		resultArray = malloc(16 * sizeof(char *));
+
+	if (resultArray == NULL)
+	{
+		write(STDOUT_FILENO,"RA_FAIL\n",8);
```
Comment (Oggal) [inline input.c]: This is a debug line ans should be removed

File: main.c [hunk 2]
```diff
@@ -12,16 +12,28 @@ int main(void)
...
 	{
+		write(STDOUT_FILENO, "->::", 4);
 		readResult = getline(&rawInput, &inputLen, stdin);
-		tokens = Prep_Input(rawInput);
+		if(!readResult > 0)
+			break;
+		tokens = Prep_Input(rawInput, tokens);
 		if(tokens == NULL || tokens[0] == NULL)
+		{
+			write(STDOUT_FILENO, "Wo\n",3);
```
Comment (Oggal) [inline main.c]: Debug line should be removed.

**Review by Oggal — COMMENTED** [review]
Need to remove debug write statements before merge

File: ops.c [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	pid_t childPID;
+	int zero = 0;
+	/* if passed NULL, Return -1 */
+	if (argV == NULL || argV[0] == NULL)
+		return (-1);
+	childPID = fork();
+	if(childPID == 0)
+	{
+		if (execve(argV[0], argV, NULL) == -1);
+		exit(errno);
```
Comment (nicholasthom1015) [inline ops.c:16]: tab over one

File: input.c [hunk 4]
```diff
@@ -19,3 +24,11 @@ char **Prep_Input(char *line)
...
         return(resultArray);
 }
+
+int Validate_Input(char **tokens)
+{
+	struct stat sb;
+	int result;
+	result = (stat(tokens[0], &sb) == 0);
+	return (result);
+}
```
Comment (nicholasthom1015) [inline input.c:33]: betty check

File: main.c [hunk 5]
```diff
@@ -12,16 +12,28 @@ int main(void)
...
+		if(Validate_Input(tokens))
+			Run_Command(tokens);
+        }while(readResult > 0);
 	printf("Goodbye!\n");
+	for (inputLen = 0; inputLen < 16; inputLen++)
+		free(tokens[inputLen]);
+	free(tokens);
+	free(rawInput);
 	return (0);
 }
```
Comment (nicholasthom1015) [inline main.c:38]: betty check

File: ops.c [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+	}
+	else
+	{
+		waitpid(childPID, NULL, WNOHANG | WUNTRACED);
+	}
+	return (0);
+
+}
```
Comment (nicholasthom1015) [inline ops.c:24]: betty check

File: shell_head.h [hunk 7]
```diff
@@ -15,9 +15,9 @@
 
 
 /* FUNCTIONS*/
+char **Prep_Input(char *, char **);
 char *_strcpy(char *dest, const char *src);
 char *_strcat(char *dest, char *src);
-char **Prep_Input(char *);
 int Validate_Input (char **);
 int Run_Command (char **);
```
Comment (nicholasthom1015) [inline shell_head.h:23]: betty check

**Review by nicholasthom1015 — CHANGES_REQUESTED** [review]
BETTY CHECK

**Pull request merged**